inc/toolbox.class.php

```diff
@@ -2753,10 +2753,12 @@ public static function removeHtmlSpecialChars($string) {
     * Slugify
     *
     * @param string $string String to slugify
+    * @param string $prefix Prefix to use (anchors cannot begin with a number)
     *
     * @return string
     */
-   public static function slugify($string) {
+   public static function slugify($string, $prefix = 'slug_') {
+      $string = str_replace('-', ' ', $string);
       $string = transliterator_transliterate("Any-Latin; NFD; [:Nonspacing Mark:] Remove; NFC; [:Punctuation:] Remove; Lower();", $string);
```

Comment:
I had best results with this.
1. It converts all chars to Latin.
2. It converts all Latin to ASCII (this removes diacritic marks and special chars that cannot be translit to ASCII).
3. It removes all chars except alphanum, `-`, `_` and `.` that are allowed in slug.
```diff
diff --git a/inc/toolbox.class.php b/inc/toolbox.class.php
index 087c856b41..11f14cf10d 100644
--- a/inc/toolbox.class.php
+++ b/inc/toolbox.class.php
@@ -2576,10 +2576,9 @@ class Toolbox {
     * @return string
     */
    public static function slugify($string) {
-      $string = transliterator_transliterate("Any-Latin; NFD; [:Nonspacing Mark:] Remove; NFC; [:Punctuation:] Remove; Lower();", $string);
+      $string = transliterator_transliterate("Any-Latin; Latin-ASCII; [^a-zA-Z0-9\.\ -_] Any-remove;", $string);
       $string = str_replace(' ', '-', self::strtolower($string, 'UTF-8'));
-      $string = self::removeHtmlSpecialChars($string);
-      $string = preg_replace('~[^0-9a-z]+~i', '-', $string);
+      $string = preg_replace('~[^0-9a-z_\.]+~i', '-', $string);
       $string = trim($string, '-');
       if ($string == '') {
          //prevent empty slugs; see https://github.com/glpi-project/glpi/issues/2946
```

Reply:
This indeed is much more better for one case in tests; but seems the « ß » character is not correctly supported (becomes "ss" instead of "sz" - the later seems better to me, but I'm not an expert at all)

Reply:
This one is a special case. It depends on following letters, but in the tested case, it should be rendered as `ss`.

See https://en.wikipedia.org/wiki/%C3%9F#Substitution_and_all_caps

Reply:
By the way, should we keep the `strtolower` call ? Using caps in slugs is not a problem.

Reply:
While this is not a problem, a case change should not change the slug (was initially used to produce anchors on the FAQ); so using lower case seems legit to me.